### PR TITLE
ci: Add codespell to CI

### PR DESCRIPTION
**Pull request by benjaminhuth**
This adds a spell checking tool to the CI along with a lot of fixes.

### CODE REVIEW

**Review by andiwand — APPROVED** [review]
looks good to me. maybe @paulgessinger wants to take a look before we merge

**Comment by paulgessinger** [comment]
Could be a good idea. Did you check this for false hits?

**Comment by paulgessinger** [comment]
Also: could we get the FPE PR in before this one?

**Review by andiwand — APPROVED** [review]

File: CI/codespell_ignore.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+parm
+writet
+localy
+lastr
+digitial
+exprot
+pring
+aline
+boxs
+ans
```
Comment (AJPfleger) [inline CI/codespell_ignore.txt:20]:
```suggestion
ans
dthe
```
should we also sort them alphabetically?

**Review by AJPfleger — CHANGES_REQUESTED** [review]
I added `dThe` to the ignore list.

Do we need to mark this PR as breaking since `auxiliary` may be part of the python API?

File: CI/check_spelling [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+codespell -S "*.ipynb,*.onnx,_build,*.svg" -I ./CI/codespell_ignore.txt ./Core ./Examples ./Tests ./Plugins ./docs
+
+RETVAL=$?
+
+if [[ $RETVAL -ne 0 ]]; then
+    echo ""
+    echo "codespell detected incorrect spelling. You have to options how to deal with this:"
```
[the remark this answers is not in the document]
Comment (benjaminhuth) [reply CI/check_spelling:9]: Haha kind of sad to have such an error in a spelling-PR :D

**Comment by benjaminhuth** [comment]
> Do we need to mark this PR as breaking since `auxiliary` may be part of the python API?

The python API is not subject of our versioning I think.

**Review by andiwand — APPROVED** [review]

**Pull request merged**